apps/wallet-mobile/src/features/Swap/SwapCard/SwapCard.tsx

```diff
@@ -0,0 +1,194 @@
+}
+
+export const SwapCard = ({label, onChange, value, wallet, amount, navigateTo, hasError}: SwapCardProp) => {
+  const strings = useStrings()
+  const {quantity, tokenId} = amount
+  const focusRef = useRef<TextInput>(null)
+
+  const tokenInfo = useTokenInfo({wallet, tokenId})
+  const noTokenSelected = tokenId === 'noTokenSelected'
+
+  const name = tokenInfo.ticker ?? tokenInfo.name
+  const denominatedQuantity = Quantities.denominated(quantity, tokenInfo.decimals ?? 0)
+
+  const focusInput = () => {
+    if (focusRef?.current) {
+      focusRef.current.focus()
+    }
+  }
+  return (
+    <>
+      <View style={[styles.container, hasError && styles.borderError]}>
+        {label != null && <Text style={[styles.label, hasError && styles.labelError]}>{label}</Text>}
+
+        <View style={styles.content}>
+          <Pressable style={styles.amountWrapper} onPress={focusInput}>
+            <AmountInput onChange={onChange} value={value} inputRef={focusRef} />
+          </Pressable>
+
+          <Spacer width={7} />
+
+          <View style={styles.rightSection}>
+            <TouchableOpacity onPress={() => navigateTo && navigateTo()}>
+              <View style={styles.sectionContainer}>
+                <Boundary loading={{fallback: <Placeholder />}} error={{fallback: () => <Placeholder />}}>
+                  {noTokenSelected ? (
+                    <Icon.Coins size={24} color={COLORS.TEXT_GRAY3} />
+                  ) : (
+                    <TokenIcon wallet={wallet} tokenId={tokenInfo.id} size="small" />
+                  )}
+                </Boundary>
+
+                <Spacer width={8} />
+
+                <Text style={styles.coinName}>{noTokenSelected ? strings.selectToken : name}</Text>
+
+                <Ionicons name="chevron-forward-outline" size={20} color="black" />
+              </View>
+            </TouchableOpacity>
+
+            <Spacer width={8} />
+
+            <Text style={styles.balanceText}>{`Current balance: ${denominatedQuantity} ADA`}</Text>
+          </View>
+        </View>
+      </View>
+
+      {hasError && (
+        <>
+          <Spacer height={4} />
+
+          <Text style={styles.errorText}>Not enough balance</Text>
+        </>
+      )}
+    </>
+  )
+}
+
+type AmountInputProps = {
+  value?: string
+  onChange(value: string): void
+  inputRef?: React.RefObject<TextInput>
+}
+const AmountInput = ({onChange, value, inputRef}: AmountInputProps) => {
+  // TODO add more formatting if is the case
+  const onChangeText = (text: string) => {
+    onChange(text)
+  }
+
+  return (
+    <TextInput
+      returnKeyType="done"
+      keyboardType="numeric"
+      autoComplete="off"
+      value={value}
+      placeholder="0"
+      onChangeText={onChangeText}
+      allowFontScaling
+      selectionColor={COLORS.TRANSPARENT_BLACK}
+      style={styles.amountInput}
+      underlineColorAndroid="tra2sparent"
+      ref={inputRef}
+    />
+  )
+}
+
+const messages = defineMessages({
```

Comment:
consume within the `common/strings`

Reply:
was thinking to move this component to the `components` folder later